datavirt64/dynamicvdb-datafederation/README.md

```diff
@@ -0,0 +1,116 @@
+
+The configuration details are passed to the pod through a secret.
+
+## Deploying the Quickstart
+
+This quickstart can be deployed as follows:
+
+1.  Instantiate one of the following templates:
+
+    * `datavirt64-basic-s2i`
+    * `datavirt64-secured-s2i` (includes configuration for https and jdbc (over SNI) routes)
+
+    Specifying parameters as follows:
+
+    * `SOURCE_REPOSITORY_URL`: https://github.com/jboss-openshift/openshift-quickstarts
+    * `SOURCE_REPOSITORY_REF`: master
+    * `CONTEXT_DIR`: datavirt64/dynamicvdb-datafederation/app
+
+    Remember to note the Teiid username and password (both randomly generated), which should be displayed after processing the template.  Optionally, you can specify your own values:
+
+    * `TEIID_USERNAME`: teiidUser
+    * `TEIID_PASSWORD`: sup3rSecret!
+
+    > Note: the password must adhere to the password strength rules or deployment will fail.
+
+2.  Create the secret for the datasource configuration.
+
+    ```
+    $ oc secrets new datavirt-app-config datasources.env
+    ```
+
+3.  Link the secret to the service account used by the application.
+
+    ```
+    $ oc secrets link datavirt-service-account datavirt-app-config
+    ```
+
+4.  Experiment with other the database types.
+
+    Add `QS_DB_TYPE` to the `datavirt-app` deployment configuration, e.g.:
+
+    ```
+    $ oc env dc/datavirt-app QS_DB_TYPE=mysql5
+    ```
+
+    Set this to one of the values listed in the _Datasource Configuration_ section.  Follow the type specific instructions for initializing the database (e.g. seeding the data).  After modifying the deployment configuration, the current deployment should be replaced with a new deployment containing your changes.
+
+> Note: Steps two and three are unnecessary if using the [datavirt-app-secret](https://github.com/jboss-openshift/application-templates/blob/master/secrets/datavirt-app-secret.yaml), as `datavirt-app-config` secret is defined and linked to the service account when that file is installed (e.g. `oc create -f application-templates/secrets/datavirt-app-secret.yaml`).
+
+### Using Red Hat JBoss Data Grid for Materialization
+
+As a prerequisite, you will need to create a data grid service in your OpenShift project.  This can
+be done by using one of the datagrid71-* templates (e.g. datagrid71-basic).  When processing the template,
+ensure the following parameters are set appropriately:
+
+* `DATAVIRT_CACHE_NAMES`=stockCache
+
+The above will configure three caches: stockCache, ST_stockCache and teiid-alias-naming-cache.
+
+Once the data grid server is up and running, a materialized view based on a JDG cache can be added to the project by specifying the `DATAGRID_MATERIALIZATION` environment variable on the build configuration, e.g.:
+
+    ```
+    $ oc env bc/datavirt-app DATAGRID_MATERIALIZATION=true
```

Comment:
New build does not cause datagrid-materialization/src/vdb/jdg-mat-cache-vdb.xml to be involved,triggered, included.. anything. How can I access it? And where to put those selects under?